### PR TITLE
Dont use api

### LINKED ISSUE #178: Cell blacklist ("don't use") API
Provide a way for technologies / project generators to blacklist the use of certain standard cells (known to cause DRC errors, etc).

Implementation detail: we can probably add it in `init_design` after running the main `init_design` or something.

### PR DESCRIPTION
**Pull request by colinschmidt**
Close #178

### CODE REVIEW

**Comment by colinschmidt** [comment]
For the record I just deleted the old hammer-robot comments about out-dated test failures

**Comment by colinschmidt** [comment]
Also this Fixes #178

File: src/hammer-vlsi/defaults.yml [hunk 1]
```diff
@@ -158,6 +158,16 @@ vlsi.inputs:
   # - "power" - This obstruction type stops only power routing/straps inside it
   placement_constraints: []
 
+  # Don't use list mode. (str)
+  # Specifies how where to get the list of cells to not use.
+  # Valid options:
+  # auto - Use the technology plugin's value for the don't use list
+  # manual - Use the list from the variable below, overriding the technology plugin
+  dont_use_mode: auto
```
[the remark this answers is not in the document]
Comment (colinschmidt) [reply src/hammer-vlsi/defaults.yml]: It would append the manual list to the technology's list?

[the remark this answers is not in the document]
Comment (colinschmidt) [reply src/hammer-vlsi/defaults.yml]: Sure seems like a good idea.

File: src/hammer-vlsi/defaults.yml [hunk 2]
```diff
@@ -158,6 +158,16 @@ vlsi.inputs:
   # - "power" - This obstruction type stops only power routing/straps inside it
   placement_constraints: []
 
+  # Don't use list mode. (str)
+  # Specifies how where to get the list of cells to not use.
```
Comment (jwright6323) [inline src/hammer-vlsi/defaults.yml]: how where

**Comment by edwardcwang** [comment]
Let's hold off on merging this for now.

File: src/hammer-vlsi/hammer_vlsi/hammer_tool.py [hunk 3]
```diff
@@ -1338,15 +1338,17 @@ def get_dont_use_list(self) -> List[str]:
 
         # dont_use_list will only be used in manual and append mode
         manual_dont_use_list = self.get_setting("vlsi.inputs.dont_use_list")  # type: List[str]
+        assert isinstance(manual_dont_use_list, list), "vlsi.inputs.dont_use_list must be a list"
 
         # tech_dont_use_list will only be used in auto and append mode
         tech_dont_use_list_raw = self.technology.config.dont_use_list  # type: Optional[List[str]]
         tech_dont_use_list = tech_dont_use_list_raw if tech_dont_use_list_raw is not None else []  # type: List[str]
+        assert isinstance(tech_dont_use_list, list), "The technology's don't use list must be a list"
```
[the remark this answers is not in the document]
Comment (colinschmidt) [reply src/hammer-vlsi/hammer_vlsi/hammer_tool.py]: Ok makes sense. Yeah

**Comment by colinschmidt** [comment]
LGTM!

**Pull request merged**